x-pack/plugins/security_solution/server/endpoint/services/agent/clients/crowdstrike/crowdstrike_agent_status_client.ts

```diff
@@ -59,6 +67,46 @@ export class CrowdstrikeAgentStatusClient extends AgentStatusClient {
     return keyBy(agentStatusResponse.data?.resources, 'id');
   }
 
+  private async getAgentDetailsFromConnectorAction(
+    agentIds: string[]
+  ): Promise<Record<string, LimitedCrowdstrikeHostDetails>> {
+    try {
+      const getAgentDetails = (await this.connectorActions.execute({
+        params: {
+          subAction: SUB_ACTION.GET_AGENT_DETAILS,
+          subActionParams: {
+            ids: agentIds,
+          },
+        },
+      })) as ActionTypeExecutorResult<CrowdstrikeGetAgentsResponse>;
+      const resources = getAgentDetails.data?.resources as Array<{
```

Comment:
adding proper type should avoid the casting here as well.

Reply:
I agree, this could be preferred. However, all CrowdStrike schemas are currently designed with `schema.maybe()` to prevent response validation errors, meaning every field is treated as `optional`. By doing this, we implicitly indicate that we expect these fields to be present. What are your thoughts on this approach?